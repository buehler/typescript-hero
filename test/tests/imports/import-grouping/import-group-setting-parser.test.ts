import {
  ImportGroupIdentifierInvalidError,
  ImportGroupSettingParser,
  KeywordImportGroup,
  RegexImportGroup,
} from '../../../../src/imports/import-grouping';
import { expect } from '../../setup';

describe('ImportGroupSettingParser', () => {

  it('should parse a simple keyword', () => {
    const result = ImportGroupSettingParser.parseSetting('Workspace') as KeywordImportGroup;

    expect(result).to.matchSnapshot();
  });

  it('should parse a simple regex', () => {
    const result = ImportGroupSettingParser.parseSetting('/foobar/') as RegexImportGroup;

    expect(result).to.matchSnapshot();
  });

  it('should parse a complex keyword pattern', () => {
    const result = ImportGroupSettingParser.parseSetting({
      identifier: 'Workspace',
      order: 'desc',
    }) as KeywordImportGroup;

    expect(result).to.matchSnapshot();
  });

  it('should parse a complex regex pattern', () => {
    const result = ImportGroupSettingParser.parseSetting({
      identifier: '/foobar/',
      order: 'desc',
    }) as RegexImportGroup;

    expect(result).to.matchSnapshot();
  });

  it('should throw on non found keyword and regex', () => {
    const fn = () => ImportGroupSettingParser.parseSetting('whatever');

    expect(fn).to.throw(ImportGroupIdentifierInvalidError);
  });

  it('should parse a regex with "or"', () => {
    const result = ImportGroupSettingParser.parseSetting('/angular|react/');

    expect(result).to.matchSnapshot();
  });

  it('should parse a regex with "@"', () => {
    const result = ImportGroupSettingParser.parseSetting('/@angular/');

    expect(result).to.matchSnapshot();
  });

  it('should parse a complex regex', () => {
    const result = ImportGroupSettingParser.parseSetting('/(@angular|react)/core/(.*)/');

    expect(result).to.matchSnapshot();
  });

});
