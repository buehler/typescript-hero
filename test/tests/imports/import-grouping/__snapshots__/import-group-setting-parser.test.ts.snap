// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportGroupSettingParser should parse a complex keyword pattern 1`] = `
KeywordImportGroup {
  "imports": Array [],
  "keyword": 2,
  "order": "desc",
}
`;

exports[`ImportGroupSettingParser should parse a complex regex 1`] = `
RegexImportGroup {
  "imports": Array [],
  "order": "asc",
  "regex": "/(@angular|react)/core/(.*)/",
}
`;

exports[`ImportGroupSettingParser should parse a complex regex pattern 1`] = `
RegexImportGroup {
  "imports": Array [],
  "order": "desc",
  "regex": "/foobar/",
}
`;

exports[`ImportGroupSettingParser should parse a regex with "@" 1`] = `
RegexImportGroup {
  "imports": Array [],
  "order": "asc",
  "regex": "/@angular/",
}
`;

exports[`ImportGroupSettingParser should parse a regex with "or" 1`] = `
RegexImportGroup {
  "imports": Array [],
  "order": "asc",
  "regex": "/angular|react/",
}
`;

exports[`ImportGroupSettingParser should parse a simple keyword 1`] = `
KeywordImportGroup {
  "imports": Array [],
  "keyword": 2,
  "order": "asc",
}
`;

exports[`ImportGroupSettingParser should parse a simple regex 1`] = `
RegexImportGroup {
  "imports": Array [],
  "order": "asc",
  "regex": "/foobar/",
}
`;
